configuration/config2/SchemaItem.cpp

```diff
@@ -479,6 +479,17 @@ std::shared_ptr<const SchemaItem> SchemaItem::getSchemaRoot() const
 }
 
 
+void SchemaItem::getPath(std::string &path) const
+{
+    path = getProperty("name") + path;
+    if (!m_pParent.expired())
```

Comment:
remind me what it means to expire in this context

Reply:
For this class, m_pParent is a std::weak_ptr. The expired method will return true if the object to which the pointer points has been deleted. Weak pointers maintain a pointer, but do not increment the reference count.